### ISSUE TITLE
Target 'make rpm' does not seem to honour cpuset flag?

### ISSUE THREAD

**Issue by jovandeginste** [adaptivecomputing/torque#369]
When I run the following commands:

```
./configure --enable-cpuset --with-default-server=xxx --disable-spool --enable-acct-x --enable-geometry-requests
make -j $(nproc)
```

The resulting binary has cpusets enabled:

```
./src/server/pbs_server --about
Package:     torque 6.0.1
Sourcedir:   /tmp/torque
Configure:    '--enable-cpuset' '--with-default-server=xxx' '--disable-spool' '--enable-acct-x' '--enable-geometry-requests'
<...>
```

However, when I (then) run `make rpm`, the binary in the rpm has no cpusets:

```
# /usr/sbin/pbs_server --about
Package:     torque 6.0.1
Sourcedir:   /root/rpmbuild/BUILD/torque-6.0.1
Configure:    '--build=x86_64-redhat-linux-gnu' '--host=x86_64-redhat-linux-gnu' '--program-prefix=' '--prefix=/usr' '--exec-prefix=/usr' '--bindir=/usr/bin' '--sbindir=/usr/sbin' '--sysconfdir=/etc' '--datadir=/usr/share' '--includedir=/usr/include' '--libdir=/usr/lib64' '--libexecdir=/usr/libexec' '--localstatedir=/var' '--sharedstatedir=/var/lib' '--mandir=/usr/share/man' '--infodir=/usr/share/info' '--includedir=/usr/include/torque' '--with-default-server=xxx' '--with-server-home=/var/spool/torque' '--without-debug' '--disable-libcpuset' '--with-sendmail=/usr/sbin/sendmail' '--disable-numa-support' '--disable-memacct' '--disable-top-tempdir-only' '--with-trqauthd-sock-dir=/tmp' '--disable-dependency-tracking' '--disable-gui' '--without-tcl' '--with-rcp=scp' '--enable-syslog' '--disable-gcc-warnings' '--disable-munge-auth' '--disable-munge-library' '--without-pam' '--disable-drmaa' '--disable-qsub-keep-override' '--disable-blcr' '--disable-cpuset' '--disable-cgroups' '--enable-spool' 'build_alias=x86_64-redhat-linux-gnu' 'host_alias=x86_64-redhat-linux-gnu' 'CXXFLAGS=-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic' 'LDFLAGS=-Wl,-z,relro ' 'CFLAGS=-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic' 'PKG_CONFIG_PATH=:/usr/lib64/pkgconfig:/usr/share/pkgconfig'
Buildcflags: -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic -D_LARGEFILE64_SOURCE 
```

The Makefile contains the following line:

```
RPM_AC_OPTS =  --without munge --with syslog --define "torque_server xxx" --with scp --without pam --without drmaa
```

(note the missing `--with cpuset` ...)

I can build manually:

```
make -j $(nproc) dist
rpmbuild --with cpuset --without munge --with syslog --define "xxx" --with scp --without pam --without drmaa -tb torque-6.0.1.tar.gz
```

After installing this rpm, I get a working binary:

```
# /usr/sbin/pbs_server --about
Package:     torque 6.0.1
Sourcedir:   /root/rpmbuild/BUILD/torque-6.0.1
Configure:    '--build=x86_64-redhat-linux-gnu' '--host=x86_64-redhat-linux-gnu' '--program-prefix=' '--prefix=/usr' '--exec-prefix=/usr' '--bindir=/usr/bin' '--sbindir=/usr/sbin' '--sysconfdir=/etc' '--datadir=/usr/share' '--includedir=/usr/include' '--libdir=/usr/lib64' '--libexecdir=/usr/libexec' '--localstatedir=/var' '--sharedstatedir=/var/lib' '--mandir=/usr/share/man' '--infodir=/usr/share/info' '--includedir=/usr/include/torque' '--with-default-server=xxx' '--with-server-home=/var/spool/torque' '--without-debug' '--disable-libcpuset' '--with-sendmail=/usr/sbin/sendmail' '--disable-numa-support' '--disable-memacct' '--disable-top-tempdir-only' '--with-trqauthd-sock-dir=/tmp' '--disable-dependency-tracking' '--disable-gui' '--without-tcl' '--with-rcp=scp' '--enable-syslog' '--disable-gcc-warnings' '--disable-munge-auth' '--disable-munge-library' '--without-pam' '--disable-drmaa' '--disable-qsub-keep-override' '--disable-blcr' '--enable-cpuset' '--disable-cgroups' '--enable-spool' 'build_alias=x86_64-redhat-linux-gnu' 'host_alias=x86_64-redhat-linux-gnu' 'CXXFLAGS=-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic' 'LDFLAGS=-Wl,-z,relro ' 'CFLAGS=-O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic' 'PKG_CONFIG_PATH=:/usr/lib64/pkgconfig:/usr/share/pkgconfig'
```

I guess this is a bug in `configure`?

**Comment by stdweird** [comment]
@jovandeginste but the output from the `--about` from the rpm has `--enable-cpuset`?

**Comment by jovandeginste** [comment]
@stdeweird the rpm created by `make rpm` does not, the one created by me (last set of commands) does...

**Comment by mej** [comment]
It's not a bug per se.  It's a lack-of-feature.  If you look at `configure.ac`, you'll see that certain `configure` arguments get added to `RPM_AC_OPTS`, but many do not.  This work has to be done by hand because there is no automatic way to determine the `rpmbuild` option which corresponds to a given `configure` argument.  So someone will need to submit a PR to add either `--with cpuset` or `--without cpuset` to `RPM_AC_OPTS` based on the outcome of the `configure.ac` block that looks at `AC_ARG_ENABLE(cpuset)` (and possibly others).

**Comment by jovandeginste** [comment]
OK fair enough

**Comment by jovandeginste** [comment]
@mej the PR is created against branch 6.0.1

**Comment by dbeer** [comment]
I believe this is now resolved since merging in pull request #370.
